### PR TITLE
Feature/wallet UI

### PR DESCRIPTION
**Pull request by DollyVolley**
## Description
Add views and flows for connecting and creating a wallet

## Improvements
Refactored Wallet mixin to fit the wallet flow functionality

## How Has This Been Tested?
Manually

## Checklist:
If you have any comments regarding one of these points just write it down.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/components/WalletModal/walletModal.ts [hunk 1]
```diff
@@ -2,41 +2,143 @@ import {Component, Mixins} from "vue-property-decorator";
...
+        SetPassword,
+        PromptPassword,
+        BalanceOverview,
+        ShowPassphrase,
+        ConfirmPassphrase,
+        DownloadKeystore,
+        EnterPassphrase,
+        Keplr
     },
     computed: mapGetters({
```
Comment (floriandulzky) [inline src/components/WalletModal/walletModal.ts:54]: empty section could be removed

File: src/components/WalletModal/walletModal.ts [hunk 2]
```diff
@@ -2,41 +2,143 @@ import {Component, Mixins} from "vue-property-decorator";
...
 
-export default class Home extends Mixins(Wallet) {
-    // Refactor
-    password: string = "";
-    newPassword: string = "";
-    localStoragePassword: string = "";
+    private errorMessage: string = ""
 
-    file: File| null = null;
+    async advanceState(result: any){
```
Comment (floriandulzky) [inline src/components/WalletModal/walletModal.ts:73]: simplify with private function calls?

File: src/mixins/Wallet.ts [hunk 3]
```diff
@@ -25,76 +30,98 @@ export default class Wallet extends Vue {
...
         } else if(!this.userwallet.initialized && !this.initializing) {
+            this.initializing = true
             await this.userwallet.initializeWallet()
             this.$store.commit("setThorchainAddress", this.userwallet.getAddressByChain("THOR"))
-            await this.getBalances()
             this.initializing = false
+        } else if (this.userwallet.initialized &&
+                   this.userwallet.type === WalletType.Keystore &&
+                   this.walletBackup !== null) {
+            console.info("Backing up keystore")
```
Comment (floriandulzky) [inline src/mixins/Wallet.ts:117]: debug output?

**Review by floriandulzky — APPROVED** [review]
Looks good for me 💯 .  
Not so easy to review because it's a big change. I commented with some questions because I am really not sure if my changes makes sense.

File: src/components/WalletModal/ConfirmPassphrase/confirmPassphrase.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        word3: HTMLInputElement[]
+    }
+
+    beforeMount() {
+        this.wordList = this.passphrase.split(" ")
+        this.getRandomWordIndices()
+    }
+
+    mounted() {
+        this.userWordList = [this.$refs.word0[0], this.$refs.word1[0], this.$refs.word2[0], this.$refs.word3[0]]
```
[the remark this answers is not in the document]
Comment (DollyVolley) [reply src/components/WalletModal/ConfirmPassphrase/confirmPassphrase.ts:27]: It is already an array, or what do you mean?
Vue Refs which are in v-for loops are retrieved as arrays, even though I specify a unique name for every input element.

File: src/components/WalletModal/FromLocalStorage/fromLocalStorage.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+    submitPassword() {
+        this.loading = true
+        this.$emit('connect-backup', this.password)
+    }
+
+    @Watch("errorMessage")
+    disableLoadingScreen() {
+        if(this.errorMessage !== "") {
+            this.loading = false
```
[the remark this answers is not in the document]
Comment (DollyVolley) [reply src/components/WalletModal/FromLocalStorage/fromLocalStorage.ts:25]: In the modal controller logic I set the error message to "" as soon as it processes input from it's views. The loading screen would not be active during the loading time therefore. 

see walletModal.ts:73 advanceState()

**Review by tomazmm — APPROVED** [review]

**Pull request merged**